pyfr/integrators/base.py

```diff
@@ -130,6 +134,26 @@ def cfgmeta(self):
         else:
             return {'config': cfg, 'config-0': cfg}
 
+    def grad_pvars(self):
```

Comment:
I'd move this functionality down into each plugin.  This way that can fetch the gradients, subset them, and then only convert the subsetted parts to primitive.

Reply:
Done for `integrate`, let me know if its ok for you and I'll move it also to `tavg`